### ISSUE TITLE
v1.4.0 Breaks Compatibility with PowerBook G4

### ISSUE THREAD

**Issue by gungwald** [AppleCommander/AppleCommander#29]
I know you're not going to like this, but I would like to humbly request that you keep supporting older machines. The highest version of Java supported by my PowerBook G4 with Mac OS X 10.4 is Java 1.5.

Unfortunately there is no way to get a newer version of Java on that machine, besides switching the OS to Linux. The IBM versions of Java use op-codes that are not supported by the G4 PowerPC CPU, so they immediately fail. The OpenJDK BSD options just don't work. 

The PowerBook G4 is still a fine machine for development work targeting an Apple II. AppleCommander 1.3.5 supported back to Java 1.3. That is a little crazy. But Java 1.5 support is a little less crazy.

Seeing that AppleCommander builds images for a vintage computer, I hope you can understand my preference for this old PowerBook G4, the last non-Intel Mac laptop.  :-)  :-)

**Comment by a2geek** [comment]
Are you looking for the command-line interface or more the GUI? 

We maybe can experiment with targeting an earlier JVM. Not really certain how/if that will work, but it may be worth a shot. This cascades into related projects as well (SHK, BASIC tokenizer, as well as the AppleSingle support). Some of those use Java 8 language features, so compiling with Java 1.5 becomes an issue.

**Comment by gungwald** [comment]
To actually answer your question, I'm only concerned about the command line interface.

**Comment by gungwald** [comment]
I was able to use retrolambda to convert the ac jar file to be java 5 compatible. I attached the script I used bye gradle or maven can be used too.

[AppleCommander-ac-1.4.0.jar.java5.zip](https://github.com/AppleCommander/AppleCommander/files/4084663/AppleCommander-ac-1.4.0.jar.java5.zip)



[retrolambda.txt](https://github.com/AppleCommander/AppleCommander/files/4084654/retrolambda.txt)

**Comment by a2geek** [comment]
Wow! I never knew anything like this existed. :joy: 
My quick google also came up with an actual Gradle plugin: https://github.com/evant/gradle-retrolambda

**Comment by gungwald** [comment]
I know. I just discovered it recently myself. It's pretty cool. And it basically worked the first time. The only problem I ran into was getting the command line jar tool to include the original manifest in the java5.jar. It's trickier than it should be. I usually use something else like Ant, Eclipse, or Maven.
    On Monday, January 20, 2020, 12:14:07 PM EST, A2 Geek <notifications@github.com> wrote:  
 
 
Wow! I never knew anything like this existed. 😂
My quick google also came up with an actual Gradle plugin: https://github.com/evant/gradle-retrolambda

**Comment by a2geek** [comment]
Looks like the Gradle plugin isn't going to work (at least not easily) with the weird Gradle build I setup for AppleCommander (which is fine). 

The plugin updates _every_ `*.class` file but _only_ those that are sourced from AppleCommander itself. The 3rd party libraries remain whatever version they were (1.8 mostly).

So I'm pulling in your script (altered to exist in a `tools` directory) since it updates/fixes the final product, including any dependencies.

**Comment by gungwald** [comment]
That is very interesting. You'll probably notice this but the script creates two temporary directories which are useful if your trying to figure out why something didn't work but probably should get automatically cleaned up if everything worked properly.
    On Monday, January 20, 2020, 03:34:12 PM EST, A2 Geek <notifications@github.com> wrote:  
 
 
Looks like the Gradle plugin isn't going to work (at least not easily) with the weird Gradle build I setup for AppleCommander (which is fine).

The plugin updates every *.class file but only those that are sourced from AppleCommander itself. The 3rd party libraries remain whatever version they were (1.8 mostly).

So I'm pulling in your script (altered to exist in a tools directory) since it updates/fixes the final product, including any dependencies.

**Comment by a2geek** [comment]
Yes, I did see that! 

I actually left it in place since I've been running it against the gradle output in the build directory. I *think* that means gradle will clean it up whenever I do a `./gradlew clean assemble` (etc).

**Comment by gungwald** [comment]
Yes, a "gradle clean" should remove everything in the build directory. That's a good point.
    On Monday, January 20, 2020, 03:57:05 PM EST, A2 Geek <notifications@github.com> wrote:  
 
 
Yes, I did see that!

I actually left it in place since I've been running it against the gradle output in the build directory. I think that means gradle will clean it up whenever I do a ./gradlew clean assemble (etc).

**Comment by a2geek** [comment]
I just released 1.6.0 and there is a [java5 jar in there](https://github.com/AppleCommander/AppleCommander/releases/download/v1-6-0/AppleCommander-ac-1.6.0-java5.jar).
Hopefully this works out!

**Comment by polluks** [comment]
Sorry, does not work:
java version "1.5.0_30"
Java(TM) 2 Runtime Environment, Standard Edition (build 1.5.0_30-b03-389-9M3425)
Java HotSpot(TM) Client VM (build 1.5.0_30-161, mixed mode, sharing)
````
java -jar ac.jar -as samples.dsk ascii <"ascii"
Exception in thread "main" java.lang.NoClassDefFoundError: java/util/Objects
	at io.github.applecommander.applesingle.AppleSingle.read(AppleSingle.java:146)
	at com.webcodepro.applecommander.ui.ac.putAppleSingle(ac.java:362)
	at com.webcodepro.applecommander.ui.ac.putAppleSingle(ac.java:354)
	at com.webcodepro.applecommander.ui.ac.main(ac.java:156)
````
